### PR TITLE
Create `.amplication/*` Folder For Packages

### LINKED ISSUE #2969: Simplifying environment variables
### Feature description

In a few of our packages we have an environment variable called `BASE_BUILDS_FOLDER` or `STORAGE_PATH`, both of which are set to `[path-to-local-folder]`.

It's recommended in the packages' README that users set it to a folder in the users' home folder (`/Users/myusername/temp`) but if the `.env` isn't modified a folder called `[path-to-local-folder]` is added to the packages' folder and is not git ignored.

This feels like an unnecessary and confusing step.

I suggest that we assign the folders to something specific, like `.base_builds_temp` or `.storage_temp`, and git ignore those folders. This simplifies the setup, removes potential confusion, and still allows for developers to move the temp files to somewhere else if they want.

### Use case

_No response_

### Are you willing to submit PR?

Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by MichaelSolati**
Issue Number: #2969

## PR Details

Sets the environment variables for temporary folders to `.amplication`. Also, git ignore them and add them to tsconfigs.

## PR Checklist
- [ ] Tests for the changes have been added
- [x] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/code_of_conduct.md) file for detailed contributing guidelines.

### CODE REVIEW

File: ee/packages/amplication-git-pull-service/.env [hunk 1]
```diff
@@ -19,11 +19,11 @@ KAFKA_CLIENT_ID=repository-pull
 KAFKA_REPOSITORY_PUSH_QUEUE="git.external.push.event.0"
 
 # Git
-STORAGE_PATH=[path-to-local-folder]
+STORAGE_PATH=.temp
```
Comment (yuval-hazaz) [inline ee/packages/amplication-git-pull-service/.env:22]: based on @arielweinberger suggestion - it should be .amplication_temp 
please see this comment on the original issue 
https://github.com/amplication/amplication/issues/2969#issuecomment-1166203987

**Review by yuval-hazaz — CHANGES_REQUESTED** [review]

**Comment by yuval-hazaz** [comment]
@tupe12334 I remember you said something about how the service is implemented and that changing the name will not work. 
Please add your comments here

@MichaelSolati @arielweinberger following Ofek's comments - please decide how to proceed with this one

**Comment by MichaelSolati** [comment]
Sorry for the delay; it's ready for review again. @yuval-hazaz @arielweinberger @tupe12334

**Comment by yuval-hazaz** [comment]
Thanks @MichaelSolati 
I had a talk with @tupe12334 a couple of days ago, and he mentioned that the current implementation of the path is absolut and not relative to the working directory, so I think it will take some more effort to make it work as expected.

@tupe12334 can you please and if true, please let us know how we should proceed from here

**Comment by tupe12334** [comment]
> I had a talk with @tupe12334 a couple of days ago, and he mentioned that the current implementation of the path is absolut and not relative to the working directory, so I think it will take some more effort to make it work as expected.
> 
> @tupe12334 can you please and if true, please let us know how we should proceed from here

The code in ```packages/amplication-server/src/core/build/utils/BuildFilesSaver/BuildFilesSaver.ts``` is saving the files in an absolute path, to save is as a relative path we can change the logic in the saving class but we need to update the logic in the loader side in the ```packages/amplication-git-pull-request-service/src/core/diff/utils/BuildPathFactory.ts``` file

**Comment by yuval-hazaz** [comment]
> The code in `packages/amplication-server/src/core/build/utils/BuildFilesSaver/BuildFilesSaver.ts` is saving the files in an absolute path, to save is as a relative path we can change the logic in the saving class but we need to update the logic in the loader side in the `packages/amplication-git-pull-request-service/src/core/diff/utils/BuildPathFactory.ts` file

Moving to relative paths may impact the prod env - so we need to carefully think about it before moving in this direction.
I would love to understand the impact of the current implementation on new contributors in the onboarding process. how clear is it for new developers to setup a local dev env to work with Amplication?

@arielweinberger @tupe12334 @abrl91 @mshidlov

**Comment by arielweinberger** [comment]
After discussing with @tupe12334 here is a good solution:

We'll write to a local `.amplication/builds` folder (at the Amplication root folder). This will enrich the experience as we don't write files to their filesystem root.

We will need to make sure we mount the EFS volume to the `.amplication/builds` folder. This way we don't affect how Production works.

@MichaelSolati Should we cake over from here, or are you interested in doing it?

It's important to make sure we update the helm configuration and docker-compose as well.

**Comment by MichaelSolati** [comment]
@tupe12334 and @arielweinberger, I don't want to be the one holding this back, if y'all have time to polish this up to land it, then please go for it. If not I'll have the time in the latter half of this week to work on it. Either way, let me know.

**Comment by tupe12334** [comment]
We need @MoFoGo to doc this change

**Comment by tupe12334** [comment]
I update this pr to meet our spec, from now on there will be a default .amplication folder that all the builds will auto save to `.amplicaiton/builds`

**Comment by arielweinberger** [comment]
Really awesome @tupe12334 well done both @MichaelSolati

**Review by arielweinberger — APPROVED** [review]

File: packages/amplication-git-pull-request-service/README.md [hunk 2]
```diff
@@ -3,7 +3,7 @@
 ## Environment Variables
 | Environment | Description | Value       |
 | ----------- | ----------- | ----------- |
-| BASE_BUILDS_FOLDER | path to a folder where your builds will be saved | 
+| BASE_BUILDS_FOLDER | path to a folder where your builds will be saved | The default path will be inside your amplication repository
```
Comment (yuval-hazaz) [inline packages/amplication-git-pull-request-service/README.md:6]:
```suggestion
| BASE_BUILDS_FOLDER | absolute path to a folder where your builds will be saved | for development purposes, leave this variable empty to use `.amplication/storage` relative to the execution folder. 
```

File: packages/amplication-server/README.md [hunk 3]
```diff
@@ -40,7 +40,7 @@ You can also use a more manual step-by-step approach to set up Amplication serve
 | SERVICE_JWT_SECRET |  JWT secret | /QN%^4uefRUR%]Ar |
 | DEFAULT_DISK | default storage disk | local |
 | LOCAL_DISK_ROOT | local storage disk root | ./artifacts |
-| BASE_BUILDS_FOLDER | path to a folder where your builds will be saved | [path-to-local-folder] for example /Users/myusername/temp |
+| BASE_BUILDS_FOLDER | path to a folder where your builds will be saved | The default path will be inside your amplication repository |
```
Comment (yuval-hazaz) [inline packages/amplication-server/README.md:43]: see comment above

File: packages/amplication-storage-gateway/README.md [hunk 4]
```diff
@@ -11,4 +11,4 @@
 | CHECK_USER_ACCESS_TOPIC | Kafka topics are the categories used to organize messages. Each topic has a name that is unique across the entire Kafka cluster | "auth.user.access" |
 | KAFKA_CLIENT_ID | A logical identifier of an application. Can be used by brokers to apply quotas or trace requests to a specific application. Example: booking-events-processor | "storage-queue-client |
 | KAFKA_GROUP_ID |  prevent collisions between Nest microservice client and server components | "storage-server-group |
-| BASE_BUILDS_FOLDER | path to a folder where your builds will be saved | [path-to-local-folder] for example /Users/myusername/temp |
+| BASE_BUILDS_FOLDER | path to a folder where your builds will be saved | The default path will be inside your amplication repository |
```
Comment (yuval-hazaz) [inline packages/amplication-storage-gateway/README.md:14]: see comment above

File: packages/amplication-storage-gateway/src/constants.ts [hunk 5]
```diff
@@ -1,3 +1,12 @@
...
 export const JWT_SECRET_KEY = "JWT_SECRET_KEY";
 export const BASE_BUILDS_FOLDER = "BASE_BUILDS_FOLDER";
 export const CHECK_USER_ACCESS_TOPIC = "CHECK_USER_ACCESS_TOPIC";
+export const DOT_AMPLICATION_FOLDER = join(
+  process.cwd(),
+  "..",
+  "..",
+  ".amplication"
+);
+export const DEFAULT_BUILDS_FOLDER = join(DOT_AMPLICATION_FOLDER, "builds");
```
Comment (yuval-hazaz) [inline packages/amplication-storage-gateway/src/constants.ts:12]: add path to storage

File: ee/packages/amplication-git-pull-service/README.md [hunk 6]
```diff
@@ -66,9 +66,8 @@ installationId: string
 | KAFKA_CONSUMER_GROUP | consumer group | git-pull-event |
 | KAFKA_CLIENT_ID | A logical identifier of an application. Can be used by brokers to apply quotas or trace requests to a specific application. Example: booking-events-processor | repository-pull |
 | KAFKA_REPOSITORY_PUSH_QUEUE | Kafka topics are the categories used to organize messages. Each topic has a name that is unique across the entire Kafka cluster | "git.external.push.event.0" |
-| STORAGE_PATH | max storage snapshot | [path-to-local-folder] for example /Users/myusername/temp |
+| STORAGE_PATH | max storage snapshot | The default path will be inside your amplication repository |
```
Comment (yuval-hazaz) [inline ee/packages/amplication-git-pull-service/README.md:69]: see other comment about the description

File: .amplication/.gitignore [hunk 7]
```diff
@@ -0,0 +1,2 @@
+builds/
+github/
```
Comment (yuval-hazaz) [inline .amplication/.gitignore:2]: storage/

**Comment by yuval-hazaz** [comment]
@MoFoGo we should document this. We can prepare a doc about the build process.
please add do your backlog and let's discuss this once you are ready

**Pull request merged**